crates/ruff_linter/src/rules/pycodestyle/overlong.rs

```diff
@@ -0,0 +1,142 @@
+        line: &Line,
+        indexer: &Indexer,
+        limit: LineLength,
+        task_tags: &[String],
+        tab_size: TabSize,
+    ) -> Option<Self> {
+        // The maximum width of the line is the number of bytes multiplied by the tab size (the
+        // worst-case scenario is that the line is all tabs). If the maximum width is less than the
+        // limit, then the line is not overlong.
+        let max_width = line.len() * tab_size.as_usize();
+        if max_width < limit.value() as usize {
+            return None;
+        }
+
+        // Measure the line. If it's already below the limit, exit early.
+        let width = Overlong::measure(line.as_str(), tab_size);
+        if width <= limit {
+            return None;
+        }
+
+        // Strip trailing pragma comments and, if necessary, re-measure the line.
+        let (line, width) = if let Some(line) = Overlong::strip_trailing(line, indexer, task_tags) {
+            // Re-measure the line.
+            let width = Overlong::measure(line.as_str(), tab_size);
+            if width <= limit {
+                return None;
+            }
+
+            (Cow::Owned(line), width)
+        } else {
+            (Cow::Borrowed(line), width)
+        };
+
+        let mut chunks = line.split_whitespace();
+        let (Some(_), Some(second_chunk)) = (chunks.next(), chunks.next()) else {
+            // Single word / no printable chars - no way to make the line shorter.
+            return None;
+        };
+
+        // Do not enforce the line length for lines that end with a URL, as long as the URL
+        // begins before the limit.
+        let last_chunk = chunks.last().unwrap_or(second_chunk);
+        if last_chunk.contains("://") {
+            if width.get() - last_chunk.width() <= limit.value() as usize {
+                return None;
+            }
+        }
+
+        // Obtain the start offset of the part of the line that exceeds the limit.
+        let mut start_offset = line.start();
+        let mut start_width = LineWidthBuilder::new(tab_size);
+        for c in line.chars() {
+            if start_width < limit {
+                start_offset += c.text_len();
+                start_width = start_width.add_char(c);
+            } else {
+                break;
+            }
+        }
+
+        Some(Self {
+            range: TextRange::new(start_offset, line.end()),
+            width: width.get(),
+        })
+    }
+
+    /// Return the range of the overlong portion of the line.
+    pub(super) const fn range(&self) -> TextRange {
+        self.range
+    }
+
+    /// Return the measured width of the line.
+    pub(super) const fn width(&self) -> usize {
+        self.width
+    }
+
+    /// Returns the width of a given string, accounting for the tab size.
+    fn measure(s: &str, tab_size: TabSize) -> LineWidthBuilder {
+        let mut width = LineWidthBuilder::new(tab_size);
+        width = width.add_str(s);
+        width
+    }
```

Comment:
How is this different from `width`? Doesn't account `Self::width` for tabs too?

Edit: This confused me. Maybe make those static functions standalone functions (considering that they don't need to access any information related to `Overlong`